sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkBatchOperation.cs

```diff
@@ -27,13 +27,15 @@ public class SparkBatchOperation : Operation<SparkBatchJob>
         private Response<SparkBatchJob> _response;
         private bool _completed;
         private RequestFailedException _requestFailedException;
+        private readonly SparkBatchOperationCompletionType _completionType;
 
-        internal SparkBatchOperation(SparkBatchClient client, ClientDiagnostics diagnostics, Response<SparkBatchJob> response)
+        internal SparkBatchOperation(SparkBatchClient client, ClientDiagnostics diagnostics, SparkBatchOperationCompletionType completionType, Response<SparkBatchJob> response)
```

Comment:
SparkBatchOperation should have a public constructor to allow access to an existing operation.  I've opened a PR to update the guidelines here: https://github.com/Azure/azure-sdk/pull/3359

Reply:
Could you please provide some details about this guideline? I cannot come up with a user scenario that SDK users want to create an `SparkBatchOperation` instance by themselves. They should always call `CreateSparkBatchJob` to get a `SparkBatchOperation`

Reply:
If someone started a job creation operation using `StartCreateSparkBatchJob`, and the creation operation took a while, there are a couple of reasons they might want to create the same operation, for example:
- If the process that called `StartCreateSparkBatchJob` crashed, and a different process needed to retrieve the created job id
- If a process wanted to check on the status of a job whose creation was started by another process

Reply:
The official guidance is now here: https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-lro-constructor

Reply:
Make sense. I will update code as suggested.

Reply:
It looks like this comment was resolved without adding the constructor ... or am I missing it?

Reply:
I will fix it tomorrow.